### PR TITLE
[MRG] Allow montage to be a str in raw.set_montage

### PR DESCRIPTION
**Pull request by cbrnr**
Which means that instead

    raw.set_montage(mne.channels.read_montage("biosemi64"))

one can simply type

    raw.set_montage("biosemi64")

### CODE REVIEW

**Comment by agramfort** [comment]
can you just update a test?

**Comment by cbrnr** [comment]
Sure, I added a test for the string argument. Strangely, `test_set_montage` fails on my local machine because `r0 = _fit_sphere(new_pos)[1]` results in an `UnboundLocalError: local variable 'iprint' referenced before assignment` - but I hope this is just my setup.

**Comment by cbrnr** [comment]
Yay, all CIs are happy 😄 !

File: mne/channels/montage.py [hunk 1]
```diff
@@ -788,6 +789,9 @@ def _set_montage(info, montage, update_ch_names=False, set_dig=True):
     -----
     This function will change the info variable in place.
     """
+    if isinstance(montage, str):  # load builtin montage
```
Comment (larsoner) [inline mne/channels/montage.py]: `string_types`

**Comment by larsoner** [comment]
@cbrnr this `iprint` bug was fixed:

https://github.com/scipy/scipy/pull/8144

Updating SciPy should fix it.

+1 for `whats_new` update

**Comment by cbrnr** [comment]
Thanks @larsoner! Should I put the `whats_new` entry in Changelog or API?

**Comment by larsoner** [comment]
Changelog. "API" is more for backward-incompatible API changes rather than all API changes.

**Comment by cbrnr** [comment]
Done.

File: mne/channels/tests/test_montage.py [hunk 2]
```diff
@@ -555,6 +555,13 @@ def test_set_montage():
     r0 = _fit_sphere(new_pos)[1]
     assert_allclose(r0, [0., -0.016, 0.], atol=1e-3)
 
+    # test if set_montage works with a string argument
+    raw1 = read_raw_fif(fif_fname)
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: can you rather update a test? just to limit testing time.

Comment (cbrnr) [reply mne/channels/tests/test_montage.py]: OK, I'll just replace it in this function then.

Comment (cbrnr) [reply mne/channels/tests/test_montage.py]: Done.

**Comment by agramfort** [comment]
thx @cbrnr

**Pull request merged**